python/torch_mlir/extras/fx_importer.py

```diff
@@ -258,6 +268,71 @@
     SYMBOLIC_TORCH_OPS = {key for key in SYMBOLIC_OP_TO_TORCH_OP}
 
 
+@dataclass
+class RangeConstraint:
+    min_val: int
+    max_val: int
+
+
+def sympy_expr_to_semi_affine_expr(
+    expr: sympy.Expr, symbols_map: Dict[str, AffineSymbolExpr]
+) -> AffineExpr:
+    """Translate sympy expressions to MLIR (semi-)affine expressions.
+
+    Recursively traverse the sympy expr AST and build the affine expr.
+    This is not a perfect translation. Sympy expressions are much more
+    expressive and not as constrained as affine (linear) expressions are.
+    However, for the most part, we don't need to support all of sympy.
+    PyTorch only uses a subset of sympy for capturing and expressing
+    symbolic shapes, and among what's supported, we expect the semi-affine
+    expressions (https://mlir.llvm.org/docs/Dialects/Affine/#semi-affine-maps)
+    to be sufficient.
+    """
+    if isinstance(expr, sympy.Symbol):
```

Comment:
Doesn't need to be for this patch, but I usually turn lookups like this into a dict keyed by type with a callback. I think there's a utility in this file that does a proper subclass map. Fine like you have it for now (I just usually try to avoid long if/else isinstance cascades)

Reply:
Thanks, that SGTM.